localstack/services/lambda_/invocation/event_manager.py

```diff
@@ -7,34 +7,238 @@
+    QueueUpdateWorker,
+    _create_message_attribute_hash,
+    to_sqs_api_message,
+)
+from localstack.services.sqs.utils import generate_message_id
 from localstack.utils.aws import dead_letter_queue
 from localstack.utils.aws.message_forwarding import send_event_to_target
+from localstack.utils.objects import singleton_factory
 from localstack.utils.strings import md5, to_str
 from localstack.utils.threads import FuncThread
-from localstack.utils.time import timestamp_millis
+from localstack.utils.time import now, timestamp_millis
 
 LOG = logging.getLogger(__name__)
 
 
-def get_sqs_client(function_version, client_config=None):
-    region_name = function_version.id.region
-    return connect_to(
-        aws_access_key_id=config.INTERNAL_RESOURCE_ACCOUNT,
-        region_name=region_name,
-        config=client_config,
-    ).sqs
+class EventQueueUpdateWorker(QueueUpdateWorker):
+    """
+    Regularly re-queues inflight and delayed messages whose visibility timeout has expired or delay deadline has been
+    reached.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.queues = []
+
+    def add_queue(self, queue: SqsQueue):
+        self.queues.append(queue)
+
+    def remove_queue(self, queue: SqsQueue):
+        self.queues.remove(queue)
+
+    def iter_queues(self) -> Iterable[SqsQueue]:
+        return iter(self.queues)
+
+
+class QueueManager:
+    queues: dict[str, StandardQueue]
+    queue_lock: threading.RLock
+    queue_update_worker: EventQueueUpdateWorker
+
+    def __init__(self):
+        self.queues = {}
+        self.queue_lock = threading.RLock()
+        self.queue_update_worker = EventQueueUpdateWorker()
+        self.queue_update_worker.start()
+
+    def get_queue(self, queue_name: str):
+        if queue_name not in self.queues:
+            raise ValueError("Queue not available")
+        return self.queues[queue_name]
+
+    def create_queue(self, queue_name: str) -> SqsQueue:
+        """
+        Creates a queue.
+        :param queue_name: Queue name, has to be unique
+        :return: Queue Object
+        """
+        with self.queue_lock:
+            if queue_name in self.queues:
+                return self.queues[queue_name]
+
+            # Truncate function name to ensure queue name limit of max 80 characters
+            queue = StandardQueue(
+                name=queue_name,
+                region="us-east-1",
+                account_id=config.INTERNAL_RESOURCE_ACCOUNT,
+            )
+            self.queues[queue_name] = queue
+            self.queue_update_worker.add_queue(queue)
+        return queue
+
+    def delete_queue(self, queue_name: str) -> None:
+        with self.queue_lock:
+            if queue_name not in self.queues:
+                raise ValueError("Queue not available")
+
+            queue = self.queues.pop(queue_name)
+            self.queue_update_worker.remove_queue(queue)
+
+
+class FakeSqsClient:
```

Comment:
Really like the idea!

One thing Id' suggest is moving this and accompanying logic to a separate file though. The event manager ideally shouldn't care about this implementation and is already fairly complex, so we should maybe try to keep the file from bloating too much.